server-state-machine.go

```diff
@@ -57,13 +60,18 @@ import (
 //  WAIT_FINISHED			RekeyIn; RekeyOut;
 //  CONNECTED					StoreTicket || (RekeyIn; [RekeyOut])
 
+// A cookie can be sent to the client in a HRR.
+// It contains two fields:
+// 1. The MintCookie: This field is used by mint itself to store the hash of initial client hello.
+// 2. The ApplicationCookie: This opaque value can be provided by the application (by setting a Config.CookieHandler)
+type cookie struct {
+	MintCookie        []byte `tls:"head=2"`
+	ApplicationCookie []byte `tls:"head=2"`
+}
```

Comment:
In NSS, we also start the negotiated parameters and double-check that they don't change